### PR TITLE
SAAS-1224 Add Web Client migration process description

### PR DESCRIPTION
**Pull request by karolrom-thales**

### CODE REVIEW

File: docs/products/access/api-reference/config-api/web-client.md [hunk 1]
```diff
@@ -93,6 +93,7 @@ JSON body parameters:
...
 | client_authentication_method     | no       | "CLIENT_SECRET_BASIC"                         | The authentication method that must be used for this client. Allowed values: `CLIENT_SECRET_BASIC`, `PKCE`, `PRIVATE_KEY_JWT` and `PUBLIC`. If no method is specified and `DEVICE_CODE` grant type is enabled, it will default to `PUBLIC`. Otherwise it will default to `CLIENT_SECRET_BASIC`. |
 | client_secret                    | depends  | "AF33E2BF29C54A4639AB…"                       | Client Secret (not returned on GET). Required when `client_authentication_method` is `CLIENT_SECRET_BASIC` or when `client_authentication_method` is not specified.                                                                                                                             |
+| hashed_client_secret             | no       | "$bcrypt$c=12$T/jBeKR…"                       | A hashed version of the secret using the PHC format (not returned on GET). For more information, see the [Web Client Migration](../../topics/web-clients/web-client-migration.md) page.
```
Comment (rvroon) [inline docs/products/access/api-reference/config-api/web-client.md:96]: Shouldn't this be a "depends" as well?

File: mkdocs.yml [hunk 2]
```diff
@@ -77,6 +77,7 @@ nav:
                   - 'Introduction': products/access/topics/web-clients/index.md
                   - 'Web client configuration': products/access/topics/web-clients/web-client-configuration.md
                   - 'Resource Owner Password Credentials': products/access/topics/web-clients/resource-owner-password-credentials.md
+                  - 'Migration of Web Clients without known plaintext secrets': products/access/topics/web-clients/web-client-migration.md
```
Comment (rvroon) [inline mkdocs.yml:80]:
```suggestion
                  - 'Migration of Web Clients with hashed secrets': products/access/topics/web-clients/web-client-migration.md
```

File: docs/products/access/topics/web-clients/web-client-migration.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+- **PBKDF2-SHA1**
+- **Argon2id**
+- **Bcrypt**
+
+#### PHC Format Explanation
+
+The PHC string format is a modular system for specifying password hash parameters in a portable and compact way. It is structured as
+follows: `$identifier$params$salt$hash`, where:
+
+- `$identifier` specifies the hashing algorithm (e.g., bcrypt, pbkdf2-sha1, argon2id).
```
Comment (czarnecki-piotr) [inline docs/products/access/topics/web-clients/web-client-migration.md:41]: `identifier` without `$`

File: docs/products/access/topics/web-clients/web-client-migration.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+def bcrypt_to_phc(bcrypt):
+    _, _, cost, salt_hash = bcrypt.split('$')
+    salt, hash = salt_hash[:22], salt_hash[22:]
+    translation_table = str.maketrans(
+        "./ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789",
+        "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/"
+    )
+    salt_standard_b64 = salt.translate(translation_table)
+    checksum_standard_b64 = hash.translate(translation_table)
+    return f"$bcrypt$c={cost}${salt_standard_b64}${checksum_standard_b64}"
```
Comment (czarnecki-piotr) [inline docs/products/access/topics/web-clients/web-client-migration.md:71]: I tested this code and works well when c>=10, when c was <10 then the bcrypt format is `$2a$08$OWv7kpMYT44JuwAzm8j0X.fnmzbgSpvlUCp6/uRpCRkgnduJieBoa`.
This code produces: `$bcrypt$c=08$QYx9mrOaV66LwyC1o+l2ZA$hpo1diUrxnWEr8BwTrETmipfwLkgDqc`
and this value is not correctly processed by Access. 
It worked correctly when I changed it to `$c=8$`

**Review by czarnecki-piotr — APPROVED** [review]

**Pull request merged**